main/.vuepress/config.js

```diff
@@ -53,12 +53,56 @@ module.exports = {
       }
     });
 
-    if (location.hash) {
-      // Re-navigate to the page target once content has loaded.
+    if (location.hash && location.hash !== '#') {
+      // Once content has loaded, re-navigate to the page target
+      // without triggering interfering router/history/scroll logic.
+      const hash = location.hash;
       fixups.set(['main'], _elems => {
-        const old = location.hash;
-        location.hash = '';
-        location.hash = old;
+        const stopPropagation = evt => {
+          evt.stopImmediatePropagation();
+
+          const props = {};
+          const proto = Object.getPrototypeOf(evt);
+          const propSource = proto === Event.prototype ? {} : proto;
+          for (const name of Object.getOwnPropertyNames(propSource)) {
+            if (name !== 'constructor') props[name] = evt[name];
+          }
+          console.log('suppress', evt.type, { __proto__: evt, ...props });
+        };
+        const stopEvents = types => {
+          const restorers = types.map(type => {
+            window.addEventListener(type, stopPropagation, true);
+            return () => window.removeEventListener(type, stopPropagation, true);
+          });
+          const passEvents = () => {
+            while (restorers.length > 0) restorers.pop()();
```

Comment:
Any reasons for `restorers.pop()` instead of `restorers.shift()`?  Maybe add a comment as to why, or why it doesn't matter.